libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -223,8 +357,262 @@ public static int getDtsFrameSize(byte[] data) {
+    int staticFieldsPresent = frameBits.readBits(1);
+    if (staticFieldsPresent == 1) {
+      nRefClockCode = frameBits.readBits(2);
+      nExSSFrameDurationCode = 512 * (frameBits.readBits(3) + 1);
+      if (frameBits.readBits(1) == 1) {
+        frameBits.skipBits(32 + 4);
+      }
+
+      int[] nuActiveExSSMask = new int[8];
+      frameBits.skipBits(3 + 3); // numAudioPresent, numAssets
+      for (int nAuPr = 0; nAuPr < numAudioPresent; nAuPr++) {
+        nuActiveExSSMask[nAuPr] = frameBits.readBits(extSSIndex + 1);
+      }
+
+      for (int nAuPr = 0; nAuPr < numAudioPresent; nAuPr++) {
+        for (int nSS = 0; nSS < extSSIndex + 1; nSS++) {
+          if (((nuActiveExSSMask[nAuPr] >> nSS) & 0x1) == 1) {
+            frameBits.skipBits(8); // nuActiveAssetMask
+          }
+        }
+      }
+
+      if (frameBits.readBits(1) == 1) { // bMixMetadataEnbl
+        frameBits.skipBits(2); // nuMixMetadataAdjLevel
+        int bits4MixOutMask = (frameBits.readBits(2) + 1) << 2;
+        int numMixOutConfigs = frameBits.readBits(2) + 1;
+        for (int ns = 0; ns < numMixOutConfigs; ns++) {
+          frameBits.skipBits(bits4MixOutMask); // nuMixOutChMask
+        }
+      }
+    }
+    for (int nAst = 0; nAst < numAssets; nAst++) {
+      int bits4ExSSFsize = (headerSizeType == 0) ? 16 : 20;
+      frameBits.skipBits(bits4ExSSFsize);
+    }
+
+    // Asset descriptor
+    int sampleRate = 0, channelCount = 0;
+    for (int nAst = 0; nAst < numAssets; nAst++) {
+      frameBits.skipBits(9 + 3); // nuAssetDescriptFsize, nuAssetIndex
+      if (staticFieldsPresent == 1) {
+        if (frameBits.readBits(1) == 1) { // bAssetTypeDescrPresent
+          frameBits.skipBits(4); // nuAssetTypeDescriptor
+        }
+        if (frameBits.readBits(1) == 1) { // bLanguageDescrPresent
+          frameBits.skipBits(24); // LanguageDescriptor
+        }
+        if (frameBits.readBits(1) == 1) { // bInfoTextPresent
+          int nuInfoTextByteSize = frameBits.readBits(10) + 1;
+          frameBits.skipBits(nuInfoTextByteSize * 8); // InfoTextString
+        }
+        frameBits.skipBits(5); // nuBitResolution
+
+        sampleRate = SAMPLE_RATE_EXTSS[frameBits.readBits(4)];
+        channelCount = frameBits.readBits(8) + 1;
+      }
+    }
+
+    // Number of audio sample in a compressed DTS frame
+    int sampleCount = nExSSFrameDurationCode * (sampleRate / REF_CLOCK_TABLE[ nRefClockCode]);
+    return new DtsFormatInfo(
+        MimeTypes.AUDIO_DTS_EXPRESS,
+        channelCount,
+        sampleRate,
+        extFrameSize,
+        sampleCount,
+        0);
+  }
+
+  /**
+   * Returns the size of frame header in a DTS-HD frame by parsing a few bytes of data
+   * (minimum 8 bytes) from the input bitstream(containing only Extension Sub-stream).
+   *
+   * @param frame The initial 8 bytes(minimum) of a DTS-HD frame(having only Extension Sub-stream).
+   * @return Size of the DTS-HD frame header in bytes.
```

Comment:
Does it mean it's always ok to pass an 8-byte array? It looks like it will read at most 55 bits, so I guess 8 bytes is safe and that is what it means.

But this documentation makes it sound like it will read at least 64 bits, and could read up to an infinite number:

Reply:
Yes I agree, I will update the documentation.

Reply:
This is done in the commit [ef41526](https://github.com/androidx/media/pull/275/commits/ef415268e215719ce385f5fb0e3149679a3bd91e)